pc2ccs/pc2v9
# I284 shadow debug support

### Description of what the PR does
  Adds use of Utilities.isDebugMode() to manage debug printing/tracing to the console; also adds some missing logging.

### Issue which the PR fixes
Fixes #284

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Windows 10, Eclipse 2019-12, Java 1.8_201

### Precise steps for _testing_ the PR
- Start a server loaded with the NAC21 CDP (available on the pc2 server machine at AWS, under ~ubuntu/current).
- Start one or more autojudges.
- Start a PC2 Admin; use it to:
  - grant the "Edit a Run" permission to the Feeder1 account
  - start the contest
- Start a Feeder1 client with the --debug option on the command line
- Select "Start Shadowing".
Verify that the shadow displays debug/trace output on the console.
